src/components/home/HomePage.js

```diff
@@ -1,15 +1,68 @@
 import React, { Component, Fragment } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+
 import { CardList } from '../Card';
 import mockArticles from '../../../mockdata/articles';
 import PopularPosts from './PopularPosts';
+import { fetchCategory } from '../../actions/articleActions';
+
+export const CardListArray = (props) => {
+  const { articlesByCategory } = props;
+  const categories = Object.entries(articlesByCategory);
+
+  // check if the categories are still loading
+  const fetchedCategories = categories.filter(categoryArray => (categoryArray[1].articles));
+  const failedCategories = categories.filter(categoryArray => (categoryArray[1].error));
+  const loading = categories.length > (fetchedCategories.length + failedCategories.length);
+  if (loading) {
+    return <i className="fas fa-spinner fa-3x fa-spin loading-icon" />;
+  }
 
-class HomePage extends Component {
-  state = {
-    articles: mockArticles
-  };
+  return categories.map((category) => {
+    if (category[1].articles) {
+      return (
+        <CardList
+        key={category[0]}
+        className="home-articles-cont"
+        title={category[0]}
+        article={category[1].articles.slice(0, 6)} />
+      );
+    }
+    return (
+      <h5 key={category[0]}>
+        <strong>{category[0]}</strong>
+        <br />
+        {category[1].error.message}
+      </h5>
+    );
+  });
+};
+
+CardListArray.propTypes = {
```

Comment:
This works, but I think it should be moved to the bottom of the file, next to HomePage.propTypes

Reply:
Done.